src/main/java/frc/robot/subsystems/Climb/IClimbIO.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.subsystems.Climb;
+
+import org.littletonrobotics.junction.AutoLog;
+
+public interface IClimbIO {
+    @AutoLog
+    public static class ClimbIOInputs {
+
+        public boolean m_leftLimitSwitchState;
+        public boolean m_rightLimitSwitchState;
+
+        public double m_leftVictorSPXOutputPercent = 0;
+        public double m_rightVictorSPXOutputPercent = 0;
+        
+    }
+
+    @AutoLog
+    public static class ClimbIOOutputs {
+
```

Comment:
I noticed that your `ClimbIOOutputs` class is empty. Are you sure there are no outputs for this subsystem at all?

Reply:
This is a great start on IO implementations, but there's some room for optimization in your outputs. As much as you can, try to avoid adding additional functions to IO interfaces beyond `setOutputs` and `getInputs`, and `stopMotors`.

Reply:
I originally had used output variables to control the motors (for all three subsystems), but I was concerned that since the motors were previously not being set periodically that they would not behave as expected when being set constantly. That's why I apted to use methods within the interface to control the motors, since that was closer to their original configuration and allowed the motors to be set only once (Although I did have concerns about not using any output variables and how that would affect the usability with AdvantageScope). Looking at it again now, I see how it would be beneficial to use output variables instead, I think I was just being more cautious than I needed to be when setting up the real implementations.

At one point, I had a stop motors output Boolean (for intake I think) which, if true, stopped the motors then set the Boolean to false. I quickly realized how janky this was and aptly removed it. Like you said, I do see the use of keeping the stopMotors method.

Thank you for looking over my code and providing feedback! I will keep your feedback in-mind in the future when working with interfaces.

Reply:
Great explanation, thank you! 

Safety is our number one concern, even before usability. I appreciate you being considerate of whether stopping the motors would truly do what we intend.

We will certainly test and evaluate the effectiveness of the `stopMotors()` function to ensure safety. Another way to ensure motors are truly stopped would be to set the motor outputs to zero in addition to calling `stopMotors()` in the subsystem. This two-pronged approach addresses the periodic outputs while also using the motor controllers' built-in stop.